### PR TITLE
feat(spans): Global config for AI model costs

### PR DESCRIPTION
**Pull request by jjbayer**
See https://github.com/getsentry/relay/pull/3554.

#skip-changelog

### CODE REVIEW

File: relay-dynamic-config/src/global.rs [hunk 1]
```diff
@@ -46,6 +46,9 @@ pub struct GlobalConfig {
     /// applying.
     #[serde(skip_serializing_if = "is_ok_and_empty")]
     pub metric_extraction: ErrorBoundary<MetricExtractionGroups>,
+
+    /// Configuration for AI span measurements.
+    pub ai_model_costs: ErrorBoundary<ai::ModelCosts>,
```
Comment (jjbayer) [inline relay-dynamic-config/src/global.rs:53]: Will add a `skip_serializing_if` here.

File: relay-dynamic-config/src/ai.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            .find(|cost| cost.matches(model_id, for_completion))
+            .map(|c| c.cost_per_1k_tokens)
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ModelCost {
+    model_id: LazyGlob,
+    for_completion: bool,
```
Comment (Dav1dde) [inline relay-dynamic-config/src/ai.rs:35]: Maybe this makes sense as an enum, but not familiar enough with AI terminology:

```rs
enum Usage {
    Completion,
    NotCompletionButTheDefaultUsecase,
    // MaybeSomeFutureUsecase,
}
```

@colin-sentry wdyt?

File: relay-dynamic-config/src/global.rs (hunk 1 above)
Comment (Dav1dde) [reply relay-dynamic-config/src/global.rs:53]: 👍

**Review by Dav1dde — APPROVED** [review]

File: relay-dynamic-config/src/ai.rs (hunk 2 above)
Comment (colin-sentry) [reply relay-dynamic-config/src/ai.rs:35]: the SDKs only send the static `prompt_tokens`, `completion_tokens`, and `total_tokens`, so it's probably fine to keep as a boolean

**Pull request merged**